src/tickit/core/simulation.py

```diff
@@ -0,0 +1,135 @@
+from tickit.core.management.event_router import InverseWiring
+from tickit.core.management.schedulers.master import MasterScheduler
+from tickit.core.state_interfaces.state_interface import get_interface
+from tickit.core.typedefs import ComponentID
+from tickit.utils.configuration.loading import read_configs
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TickitSimulation:
+    """A Tickit Simulation runner.
+
+    Takes a master scheduler and componenents. When the run method is called
+    scheduler and/or component tasks are created and awaited on.
+    """
+
+    _backend: str
+    _scheduler: Optional[MasterScheduler]
+    _components: Optional[Dict[ComponentID, Component]]
+
+    def __init__(
+        self,
+        backend,
+        scheduler,
+        components,
+    ) -> None:
+        self._backend = backend
+        self._scheduler = scheduler
+        self._components = components
+
+    async def run(self) -> None:
+        """Awaits the scheduler and/or component tasks indefinitely."""
+        tasks = list(self._start_tasks())
+        if tasks:
+            await asyncio.wait(tasks)
+
+    def _start_tasks(self) -> Iterable[asyncio.Task]:
+        yield from self._start_scheduler_tasks()
+        yield from self._start_component_tasks()
+
+    def _start_scheduler_tasks(self) -> Iterable[asyncio.Task]:
+        if self._scheduler is not None:
+            yield asyncio.create_task(self._scheduler.run_forever())
+
+    def _start_component_tasks(self) -> Iterable[asyncio.Task]:
+        if self._components is not None:
+            for component in self._components.values():
+                yield (
+                    asyncio.create_task(
+                        component.run_forever(*get_interface(self._backend))
+                    )
+                )
+
+
+class TickitSimulationBuilder:
+    """A Ticket simulation builder.
+
+    A utility class that uses a config file to create a scheduler and the components
+    for a simulation.
+    """
+
+    _backend: str
+    _config_path: str
+
+    _include_schedulers: bool
+    _include_components: bool
+    _components_to_run: Set[ComponentID]
+
+    def __init__(
+        self,
+        config_path: str,
+        backend: str = "internal",
+        include_schedulers: bool = True,
+        include_components: bool = True,
+        components_to_run: Optional[Set[ComponentID]] = None,
+    ) -> None:
+        """Instantiate the builder class variables.
+
+        Args:
+            config_path (str): The path to the configuration file.
+            backend (str): The message broker to be used.
+            include_schedulers (bool): A flag to determine if the master scheduler is
+                included in the simulation. Defaults to True.
+            include_components (bool): A flag to determine if components are included
+                in the simulation. Defaults to True.
+            components_to_run (Optional[Set[ComponentID]): A subset of components in
```

Comment:
Might be worth mentioning that an empty set will also result in all components being run. I'm not sure why you'd want to run none of the configured things but it'd be good to be clear that it will always run something.

Reply:
I think empty set will result in no components, run, `None` will result in all components, but that's still not ideal.
Could make `components_to_run` of type `Union[Set[ComponentID], Literal["ALL"]]` or something?

Reply:
As written and empty set and `None` do the same thing

Reply:
You're right, my bad! I don't think we should have that.

Reply:
Keep the field<sup>*</sup> as Optional and use None as the 'run everything' case?

<sup>\* I still think this should be a function not a class</sup>

Reply:
I added a clarification in the update.

Reply:
Also this is a function now, do not worry.

Reply:
However, the wierd None empty set stuff is still there. If there is a nicer solution you can think of i'll be happy to use it. I just want to have the following functionality:
 - Not specifying `components_to_run` should run all the components from the config file
 - Specifying `components_to_run` will only run the subset specified
 - To not run any components, the flag `include_components` needs to be set to False

Reply:
Remove the `components_to_run = components_to_run or set()` line and change 
```diff
-run_all = not components_to_run
+run_all = components_to_run is None
```

Reply:
I have changed it so you need to specify components, and if you want all of them you use `all`.